src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -705,10 +702,21 @@ public void doConfigSubmit(StaplerRequest req, StaplerResponse rsp) throws IOExc
             if (namingStrategy.isForceExistingJobs()) {
                 namingStrategy.checkName(name);
             }
-            FormApply.success(".").generateResponse(req, rsp, this);
+            FormApply.success(getSuccessfulDestination()).generateResponse(req, rsp, this);
         }
     }
 
+    /**
+     * Where user will be redirected after creating a {@code AbstractFolder}.
```

Comment:
(And if [JENKINS-18681](https://issues.jenkins-ci.org/browse/JENKINS-18681) were implemented, this would be even more misleading, as for regular `Folder`s you would not be sent to `configure` at all during creation.)